data-prepper-expression/src/main/antlr/DataPrepperExpression.g4

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+grammar DataPrepperExpression;
+
+@header {
+    package org.opensearch.dataprepper.expression.antlr;
+}
+
+fragment
+Digit
+    : '0'
+    | NonZeroDigit
+    ;
+
+fragment
+NonZeroDigit
+    : [1-9]
+    ;
+
+Integer
+    : '0'
+    | NonZeroDigit Digit*
+    ;
+
+Float
+    : NonZeroDigit? Digit '.' Digit
+    | NonZeroDigit? Digit '.' Digit+ NonZeroDigit
+    | '.' Digit
+    | '.' Digit* NonZeroDigit
+    ;
+
+Boolean
+    : TRUE
+    | FALSE
+    ;
+
+fragment
+StringCharacters
+    : StringCharacter+
+    ;
+
+fragment
+StringCharacter
+    : ~["\\$]
+    | EscapeSequence
+    ;
+
+fragment
+EscapeSequence
+    : '\\' [btnfr"'\\$]
+    ;
+
+fragment
+JsonPointerEscapeSequence
+    : '\\' [btnfr"'\\/]
+    ;
+
+fragment
+JsonPointerStringCharacters
+    : JsonPointerStringCharacter+
+    ;
+
+fragment
+JsonPointerStringCharacter
+    : ~["\\]
+    | JsonPointerEscapeSequence
+    ;
+
+fragment
+JsonPointerCharacter
+    : [A-Za-z0-9_]
```

Comment:
Should we support all the allowed characters that are valid here (https://github.com/opensearch-project/data-prepper/blob/32c83db5174fc81926cba2a68b832e35718e9daa/data-prepper-api/src/main/java/com/amazon/dataprepper/model/event/JacksonEvent.java#L302), or is that covered?

Reply:
Good question, this one has popped up a few times. For shorthand JsonPointers a limited character set is recognised to reduce collision. Keep in mind while space is insignificant. `/hello==/world` is valid.

An expanded JsonPointer character set is supported by the EscapedJsonPointer token rule. If you have ideas for other characters that should be covered in the shorthand notation add a comment here or in #1005